app/controllers/sub_space_booking_controller.rb

```diff
@@ -84,16 +149,65 @@ def bookings
           end
         end
     end
-
     render json: @bookings
   end
 
   def create
+    unless signed_in?
+      redirect_to login_path, alert: "You must be logged in to view this page."
+      return
+    end
+    unless current_user.booking_approval || current_user.admin?
+      redirect_to root_path, alert: "You must be approved to book spaces."
+      return
+    end
+
+    if params[:sub_space_booking][:recurring].present?
+      if params[:sub_space_booking][:recurring] == true
+        if params[:sub_space_booking][:recurring_end].present? &&
+             params[:sub_space_booking][:recurring_frequency].present?
+          if params[:sub_space_booking][:recurring_frequency] == "weekly"
+            start_time = params[:sub_space_booking][:start_time].to_datetime
+            end_time = params[:sub_space_booking][:end_time].to_datetime
+            end_date = params[:sub_space_booking][:recurring_end].to_date
+            book(params)
+            while start_time < end_date
+              params[:sub_space_booking][:start_time] = start_time + 7.days
+              params[:sub_space_booking][:end_time] = end_time + 7.days
+              start_time = params[:sub_space_booking][:start_time].to_datetime
+              end_time = params[:sub_space_booking][:end_time].to_datetime
+              book(params)
+            end
+          elsif params[:sub_space_booking][:recurring_frequency] == "monthly"
+            start_time = params[:sub_space_booking][:start_time].to_datetime
+            end_time = params[:sub_space_booking][:end_time].to_datetime
```

Comment:
Maybe merge the monthly and weekly together with a condition for the start and end times and save the result in a variable that could be used, it would save the duplication of lines